WIP: Add hasProperty() assertion

seansellek opened the pull request:
This takes the first steps towards addressing #38 by adding a base `.hasProperty()` assertion as mentioned by @Turbo87.

The assertion supports providing a custom message to facilitate building shortcuts such as `.isChecked()`.

Resolves #55

seansellek commented:
There seems to be a permissions issue surrounding Chrome in CI 😥

`You need to make sure that  /opt/google/chrome/chrome-sandbox is owned by root and has mode 4755`

Turbo87 commented:
@seansellek I'm currently on vacation, I'll have a look once I'm back (7 days)

Turbo87 commented:
@seansellek 👋  we've just merged https://github.com/simplabs/qunit-dom/pull/48, which should fix the Chrome issue. Can you rebase this PR?

Turbo87 commented on .gitignore:
```diff
@@ -16,6 +16,7 @@
 npm-debug.log*
 yarn-error.log
 testem.log
+.vscode/
```
This stuff should be move to your global `.gitignore` file (see https://help.github.com/articles/ignoring-files/)

Only project specific files should be added to this file, but no individual editor config files etc.

seansellek commented:
@Turbo87 sorry it took a bit for me to get back to this. Rebased, removed the .gitignore commit, and fixed a bug in the tests, getting coverage up to 100%.

Turbo87 commented on lib/assertions.js:
```diff
@@ -72,6 +73,23 @@ export default class DOMAssertions {
+   * if the property `value` matches the provided `value`.
+   *
+   * @param {string} property
+   * @param {object?} options
```
I'm wondering if we should mimic the `hasAttribute` API instead which accepts `name, value?, message?` 🤔

scalvert replied:
+1 to this. There's a lot to be said for the API being consistent for like methods/functions.

Turbo87 commented on API.md:
````diff
@@ -495,3 +495,18 @@ Assert that the `value` property of an [HTMLInputElement][] is empty.
+**Examples**
+
+```javascript
+assert.dom('input#agreedToTOS').hasProperty('checked', { value: true });
````
as mentioned before, I think this should mimic the API that we have for `hasAttribute()`

scalvert commented:
Thanks for pinging on this, @Turbo87. I've been pushing on it over the last little while, but haven't had as much time to get it through.

I've aligned it more to the `hasAttribute` signature, and am just working through the tests. I'll try to get this finished up over the next little while.

Turbo87 commented:
sweet, thanks for working on it :)

Turbo87 commented:
closing due to inactivity
